File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Temperature IR Bricklet 2.0

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletTemperatureIRV2 tir =
		  new BrickletTemperatureIRV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current ambient temperature (unit is °C/10)
		short ambientTemperature = tir.GetAmbientTemperature();
		Console.WriteLine("Ambient Temperature: " + ambientTemperature/10.0 + " °C");

		// Get current object temperature (unit is °C/10)
		short objectTemperature = tir.GetObjectTemperature();
		Console.WriteLine("Object Temperature: " + objectTemperature/10.0 + " °C");

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleSimple.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletTemperatureIRV2;

public class ExampleSimple {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Temperature IR Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletTemperatureIRV2 tir =
		  new BrickletTemperatureIRV2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current ambient temperature
		int ambientTemperature = tir.getAmbientTemperature(); // Can throw com.tinkerforge.TimeoutException
		System.out.println("Ambient Temperature: " + ambientTemperature/10.0 + " °C");

		// Get current object temperature
		int objectTemperature = tir.getObjectTemperature(); // Can throw com.tinkerforge.TimeoutException
		System.out.println("Object Temperature: " + objectTemperature/10.0 + " °C");

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
